netlify/helpers/github.ts

```diff
@@ -0,0 +1,480 @@
+        }, error: ${error.message}`
+      );
+    });
+
+    // create hash map of milestones
+    // because id is not guaranteed to be the same for everyone
+    userMilestones.forEach((milestone) => {
+      if (milestone?.title) {
+        this.milestoneMap[milestone.title] = milestone.number;
+      }
+    });
+
+    const newMilestones: Milestone[] = [];
+
+    // create labels and milestones that don't exist
+    const repoMilestones = await this.getMilestonesForRepo(
+      sourceRepoName
+    ).catch((error) => {
+      throw new Error(
+        `Could not get milestones for ${this.login}/${sourceRepoName}, error: ${error.message}`
+      );
+    });
+
+    repoMilestones.forEach((milestone) => {
+      if (milestone?.title) {
+        if (!this.milestoneMap[milestone.title]) {
+          newMilestones.push(milestone);
+        }
+      }
+    });
+
+    const uniques = _.uniqWith<Milestone>(newMilestones, _.isEqual);
+
+    const promises = uniques.map((milestone) => {
+      if (!milestone || !milestone.title) {
+        return;
+      }
+      return this.octokit.rest.issues
+        .createMilestone({
+          owner: this.login,
+          repo: config().defaultRepo,
+          title: milestone.title as string,
+          description: milestone.description as string,
+          state: "open",
+          due_on: milestone.due_on || undefined,
+        })
+        .then((res) => {
+          this.milestoneMap[milestone.title] = res.data.number;
+        })
+        .catch((error) => {
+          throw new Error(
+            `Could not create milestone ${milestone.title}, error: ${error.message}`
+          );
+        });
+    });
+
+    for (const milestone of uniques) {
+      if (!milestone || !milestone.title) {
+        continue;
+      }
+      const { data } = await this.octokit.rest.issues
+        .createMilestone({
+          owner: this.login,
+          repo: config().defaultRepo,
+          title: milestone.title as string,
+          description: milestone.description as string,
+          state: "open",
+          due_on: milestone.due_on || undefined,
+        })
+        .catch((error) => {
+          throw new Error(
+            `Could not create milestone ${milestone.title}, error: ${error.message}`
+          );
+        });
+
+      this.milestoneMap[milestone.title] = data.number;
+    }
+    return true;
+  }
+
+  async setupDefaultProject(): Promise<string> {
+    const {
+      user: {
+        projectsV2: {
+          nodes: [project],
+        },
+      },
+    } = await this.octokit
+      .graphql<GraphQlQueryResponseData>(getUserProjectsV2Query, {
+        login: this.login,
+      })
+      .catch((error) => {
+        throw new Error(
+          `Could not get projects for ${this.login}, error: ${error.message}`
+        );
+      });
+
+    if (!project?.id) {
+      throw new Error(
+        "No project found that matches 'coursework planner' query, please clone the project template provided."
```

Comment:
I'm not quite sure as a user what this is asking me to do... Can you clarify?

It looks like you're looking for me to go to https://curriculum.codeyourfuture.io/fundamentals/prep/ and follow the "Set up Planner" instructions? Shall we link there?

Reply:
Thanks, I will add the link